server/src/main/java/org/opensearch/client/node/NodeClient.java

```diff
@@ -107,6 +116,9 @@ public <Request extends ActionRequest, Response extends ActionResponse> Task exe
         Request request,
         ActionListener<Response> listener
     ) {
+        final SpanScope spanScope = tracer.startSpan(SPAN_NAME_PREFIX_TRANSPORT_ACTION + action.name());
```

Comment:
This is not the right please to instrument I think: it should be done in the TransportAction::execute

Reply:
I am also skeptical about adding the instrumentation at this place. But all the calls are not happening via client so will remove this. TransportAction::execute anyways I am planning to do in the other PR as it require changes for passing the tracer to TransportAction from all the concrete implementation.

Reply:
Changes for TransportAction::execute already tracking here - https://github.com/opensearch-project/OpenSearch/issues/8553